pkg/controller/virtinformers.go

```diff
@@ -165,6 +166,9 @@ type KubeInformerFactory interface {
 	// Watches for the export route config map
 	ExportRouteConfigMap() cache.SharedIndexInformer
 
+	// Watches for cpu model config map
+	CpuModelToFeaturesToDisableConfigMap() cache.SharedIndexInformer
```

Comment:
That's a valid point. We chose to use configMap instead of adding it to KubevirtCR because:

- Typically, admins shouldn't use it unless there's a Libvirt/QEMU bug or a CPU vulnerability.
- When this ConfigMap is used, it's only temporary until a vulnerability/bug is fixed in Libvirt/QEMU.
- We need to be cautious about adding complexity to the KubevirtCR API, which is already complex due to its many configuration options.
- Removing the ConfigMap option won't disrupt backward compatibility, unlike removing a new field from KubevirtCR.

Reply:
> Typically, admins shouldn't use it unless there's a Libvirt/QEMU bug or a CPU vulnerability.

I am not sure that I understand how this may affect the choice of config map vs. CR.

> When this ConfigMap is used, it's only temporary until a vulnerability/bug is fixed in Libvirt/QEMU.

We also consider this feature as a mitigation for CPU vulnerabilities (so it's not only about libvirt/qemu). Not to say that users may have to stay with one version of KubeVirt for a long time (consuming a bugfix from libvirt requires a new release of KubeVirt). Hence, the "temporary" may likely become "permanent" setting.

> We need to be cautious about adding complexity to the KubevirtCR API, which is already complex due to its many configuration options.

If this is the case, and we do not want to add any more settings to the CR, then IMHO we first need to define a strategy on how we proceed further and agree it in the community. Otherwise, we end up with cluster-wide settings scattered across various resources (CRs, config maps, etc.) and it may become a mess.

> Removing the ConfigMap option won't disrupt backward compatibility, unlike removing a new field from KubevirtCR.

I tend to disagree here. This is a user-facing feature which affects the workloads. We cannot simply add it in one minor release and then silently drop in some other. If at some point, after the release, the config map is removed, it will be the same compatibility problem since some users may start using it. IMHO, ideally, it should pass all the stages from alpha, beta and then to GA irrespective of whether it is a config map or CR setting.

Reply:
You are correct; it will increase the complexity. I modified the PR to set the configuration in kubevirtCR instead.